workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ScreenOverlayDialogFactory.kt

```diff
@@ -109,42 +103,33 @@ public open class ScreenOverlayDialogFactory<S : Screen, O : ScreenOverlay<S>>(
     initialEnvironment: ViewEnvironment,
     context: Context
   ): ScreenViewHolder<S> {
-    return viewFactory
-      .startShowing(initialContent, initialEnvironment, context)
+    return viewFactory.startShowing(initialContent, initialEnvironment, context)
   }
 
   /**
    * Build the [Dialog] for the [content] that was just created by [buildContent].
-   * Open to allow customization, typically theming.
-   *
-   * The default implementation delegates all work to the provided [Dialog.setContent]
-   * extension function. Subclasses need not call `super`.
+   * Open to allow customization, typically theming, subclasses need not call `super`.
+   *  - Note that the default implementation calls the provided [Dialog.setContent]
+   *    extension for typical setup.
+   *  - Be sure to call [ScreenViewHolder.show] from [OverlayDialogHolder.runner].
    */
-  public open fun buildDialogWithContent(content: ScreenViewHolder<S>): Dialog {
-    return Dialog(content.view.context).also { it.setContent(content) }
+  public open fun buildDialogWithContent(
+    initialRendering: O,
+    initialEnvironment: ViewEnvironment,
+    content: ScreenViewHolder<S>
+  ): OverlayDialogHolder<O> {
+    return OverlayDialogHolder(
+      initialEnvironment, Dialog(content.view.context).also { it.setContent(content) }
```

Comment:
nit:
```suggestion
      initialEnvironment, Dialog(content.view.context).apply { setContent(content) }
```
I think of `also` as side effects wrt to the receiver, whereas `apply` is explicitly about the receiver.